src/focus_manager/txm_platform.js

```diff
@@ -446,29 +446,33 @@ export class TXMPlatform {
             self.name = "FireTV";
 
             // From: https://developer.amazon.com/docs/fire-tv/identify-amazon-fire-tv-devices.html
-            const modelMatch = userAgent.match(/\bAFT[A-Z]+\b/);
+            const modelMatch = userAgent.match(/\bAFT[A-Z0-9]+\b/);
             const modelId = modelMatch && modelMatch[0];
-            var model = "Fire TV";
-            if (modelId == "AFTN") {
-                model = "Fire TV (Gen 3)";
-            } else if (modelId == "AFTS") {
-                model = "Fire TV (Gen 2)";
-            } else if (modelId == "AFTB") {
-                model = "Fire TV (Gen 1)";
-            } else if (modelId == "AFTT") {
-                model = "Fire TV Stick (Gen 2)";
-            } else if (modelId == "AFTM") {
-                model = "Fire TV Stick (Gen 1)";
-            } else if (modelId == "AFTMM") {
-                model = "Fire TV Stick 4K";
-            } else if (modelId == "AFTRS") {
-                model = "Fire TV Edition";
-            } else if (modelId == "AFTA") {
-                model = "Fire TV Cube (Gen 1)";
-            } else if (modelId == "AFTR") {
-                model = "Fire TV Cube (Gen 2)";
-            }
-            self.model = model;
+
+            // Derived from https://developer.amazon.com/docs/fire-tv/identify-amazon-fire-tv-devices.html
+            // NOTE: watch out for duplicates!
+            const knownModels = {
+                AFTA: "Fire TV Cube (Gen 1)",
+                AFTB: "Fire TV (Gen 1)",
+                AFTBAMR311: "Fire TV Edition - Toshiba HD (2018-2020)",
+                AFTDCT31: "Fire TV Edition - 4K UHD (2020)",
+                AFTEAMR311: "Fire TV Edition - Insignia HD (2018-2020)",
```

Comment:
interesting that you include the year. But wouldn't 2021 destroy all logic?
Fire TV Edition - Insignia HD (2018-2020) != Fire TV Edition - Insignia HD (2018-2021)

Why don't we include only the release year?

Reply:
I am not interpreting the model names, just using them as is, as per the official source.

If they are different names, then they are different hardware, with potentially different performance.

I would expect new models to in fact just say 2021, etc.

Reply:
right, I understand that too. My point is if they did not update the model next year-- thus, 
`Fire TV Edition - Insignia HD (2018-2020)`
becomes
`Fire TV Edition - Insignia HD (2018-2021)`

unless we are expecting people not to compare these strings, ([which we are comparing](https://github.com/socialvibe/container_core/pull/297/files#diff-100f48e290198cc1d1de1ccef536ae59R185))
it would be best to try to make sure these name won't change and they are final

But not a huge issue.